test_data/asm/expand_fixed.asm

```diff
@@ -0,0 +1,101 @@
+use std::prover::degree;
+use std::check::assert;
+
+// A term in an array expression
+enum ArrayTerm {
+    Repeat(int[]),
+    Once(int[])
+}
+
+// returns the total size of the repeated array in this array expression
+let solve: ArrayTerm[] -> int = |terms| {
+    let (_, total_size_of_non_repeated) = std::array::fold(terms, (false, 0), |(found_repeated, len), term| {
+        match term {
+            // we can have `[]*` as many times as we want
+            ArrayTerm::Repeat([]) => (found_repeated, len),
+            ArrayTerm::Repeat(a) => {
+                if found_repeated {
+                    std::check::panic("unsolvable because two terms are repeated")
+                } else {
+                    (true, len)
+                }
+            },
+            ArrayTerm::Once(a) => (found_repeated, len + std::array::len(a))
+        }
+    });
+    //std::check::assert(total_size_of_non_repeated <= degree(), || "non repeated array terms do not fit in degree");
+    degree() - total_size_of_non_repeated
+};
+
+// returns a function representing the array expression
+let expand: ArrayTerm[] -> (int -> int) = |terms| {
+    // get the total size of the repeated term
+    let size_of_repeated = solve(terms);
+    |i| {
+        let (_, res) = std::array::fold(terms, (0, 0), |(offset, res), term| {
+            let (a, len) = match term {
+                ArrayTerm::Repeat(a) => (a, size_of_repeated),
+                ArrayTerm::Once(a) => (a, std::array::len(a))
+            };
+
+            let index = i - offset;
+
+            (
+                offset + len,
+                if 0 <= index && index < len {
+                    res + a[index % std::array::len(a)]
+                } else {
+                    res
+                }
+            )
+        });
+        res
+    }
+};
+
+let repeat: int[] -> ArrayTerm = |a| ArrayTerm::Repeat(a);
+
+let once: int[] -> ArrayTerm = |a| ArrayTerm::Once(a);
+
+// test utility
+let assert_eq: expr, expr -> () = constr |left, right| {
+    let diff;
+    diff = left - right;
+    diff = 0;
+};
```

Comment:
Would change this so that the check does not rely on the witness generator:
```suggestion
let assert_eq: (int -> int), int[] -> () = |fun, expected| {
    let _ = std::array::new(degree(), |i| assert_eq(fun(i), expected[i]))
};
```